sql-plugin/src/main/java/com/nvidia/spark/rapids/RapidsHostColumnVectorCore.java

```diff
@@ -0,0 +1,139 @@
+
+/*
+ * Copyright (c) 2020, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.nvidia.spark.rapids;
+
+import org.apache.spark.sql.types.DataType;
+import org.apache.spark.sql.types.Decimal;
+import org.apache.spark.sql.vectorized.ColumnVector;
+import org.apache.spark.sql.vectorized.ColumnarArray;
+import org.apache.spark.sql.vectorized.ColumnarMap;
+import org.apache.spark.unsafe.types.UTF8String;
+
+
+/**
+ * A GPU accelerated version of the Spark ColumnVector.
+ * Most of the standard Spark APIs should never be called, as they assume that the data
+ * is on the host, and we want to keep as much of the data on the device as possible.
+ * We also provide GPU accelerated versions of the transitions to and from rows.
+ */
+public final class RapidsHostColumnVectorCore extends ColumnVector {
+
+  private final ai.rapids.cudf.HostColumnVectorCore cudfCv;
+
+  /**
+   * Sets up the data type of this column vector.
+   */
+  RapidsHostColumnVectorCore(DataType type, ai.rapids.cudf.HostColumnVectorCore cudfCv) {
+    super(type);
+    // TODO need some checks to be sure everything matches
```

Comment:
Is this going to be a follow on issue? I agree if we don't have a one to one mapping between types it would be good to have an assertion that can verify that the cudf data passed in can satisfy the requirements for the spark type.

Reply:
https://github.com/NVIDIA/spark-rapids/issues/1002